### PR TITLE
Add Create HanaInstance API

### PR DESCRIPTION
**Pull request by lagalbra**
### Latest improvements:
<i>MSFT employees can try out our new experience at <b>[OpenAPI Hub](https://aka.ms/openapiportal) </b> - one location for using our validation tools and finding your workflow. 
</i><br>
### Contribution checklist:
- [x] I have reviewed the [documentation](https://github.com/Azure/azure-rest-api-specs#basics) for the workflow.
- [x] [Validation tools](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/swagger-checklist.md#validation-tools-for-swagger-checklist) were run on swagger spec(s) and have all been fixed in this PR.
- [ ] The [OpenAPI Hub](https://aka.ms/openapiportal) was used for checking validation status and next steps.
### ARM API Review Checklist
- [x] Service team MUST add the "WaitForARMFeedback" label if the management plane API changes fall into one of the below categories. 
- adding/removing APIs.
- adding/removing properties.
- adding/removing API-version. 
- adding a new service in Azure.

Failure to comply may result in delays for manifest application. Note this does not apply to data plane APIs.
- [ ] If you are blocked on ARM review and want to get the PR merged urgently, please get the ARM oncall for reviews (RP Manifest Approvers team under Azure Resource Manager service) from IcM and reach out to them. 
Please follow the link to find more details on [API review process](https://armwiki.azurewebsites.net/rp_onboarding/ResourceProviderOnboardingAPIRevieworkflow.html).

### CODE REVIEW

**Review by kpajdzik — APPROVED** [review]

**Review by dwj300 — CHANGES_REQUESTED** [review]

File: specification/hanaonazure/resource-manager/Microsoft.HanaOnAzure/preview/2017-11-03-preview/examples/HanaInstances_List.json [hunk 1]
```diff
@@ -38,8 +38,11 @@
               "osProfile": {
                 "computerName": "myComputerName1",
                 "osType": "SUSE",
-                "version": "12 SP1"
-              }
+                "version": "12 SP1",
+                "sshPublicKey": "AAAAB3NzaC1yc2EAAAABJQAAAQB/nAmOjTmezNUDKYvEeIRf2YnwM9/uUG1d0BYsc8/tRtx+RGi7N2lUbp728MXGwdnL9od4cItzky/zVdLZE2cycOa18xBK9cOWmcKS0A8FYBxEQWJ/q9YVUgZbFKfYGaGQxsER+A0w/fX8ALuk78ktP31K69LcQgxIsl7rNzxsoOQKJ/CIxOGMMxczYTiEoLvQhapFQMs3FL96didKr/QbrfB1WT6s3838SEaXfgZvLef1YB2xmfhbT9OXFE3FXvh2UPBfN+ffE7iiayQf/2XR+8j4N4bW30DiPtOQLGUrH1y5X/rpNZNlWW2+jGIxqZtgWg7lTy3mXy5x836Sj/6L"
+              },
+              "partnerNodeId": "",
```
[the remark this answers is not in the document]
Comment (lagalbra) [reply specification/hanaonazure/resource-manager/Microsoft.HanaOnAzure/preview/2017-11-03-preview/examples/HanaInstances_List.json]: Done

**Comment by lagalbra** [comment]
As noted in https://github.com/Azure/azure-rest-api-specs/pull/5976, I have marked this API as asynchronous.

File: specification/hanaonazure/resource-manager/Microsoft.HanaOnAzure/preview/2017-11-03-preview/hanaonazure.json [hunk 2]
```diff
@@ -194,6 +194,54 @@
...
+        "operationId": "HanaInstances_Create",
+        "x-ms-long-running-operation": true,
+        "summary": "Creates a SAP HANA instance.",
+        "description": "Creates a SAP HANA instance for the specified subscription, resource group, and instance name.",
+        "x-ms-examples": {
+          "Get properties of a HANA instance": {
+            "$ref": "./examples/HanaInstances_Create.json"
+          }
+        },
+        "parameters": [
```
Comment (ravbhatnagar) [inline specification/hanaonazure/resource-manager/Microsoft.HanaOnAzure/preview/2017-11-03-preview/hanaonazure.json]: request body?

**Review by kpajdzik — APPROVED** [review]

File: specification/hanaonazure/resource-manager/Microsoft.HanaOnAzure/preview/2017-11-03-preview/hanaonazure.json (hunk 2 above)
Comment (lagalbra) [reply specification/hanaonazure/resource-manager/Microsoft.HanaOnAzure/preview/2017-11-03-preview/hanaonazure.json]: My bad; added to spec, and example updated.

Comment (lagalbra) [reply specification/hanaonazure/resource-manager/Microsoft.HanaOnAzure/preview/2017-11-03-preview/hanaonazure.json]: @ravbhatnagar Please review

**Comment by ravbhatnagar** [comment]
Signing off frm ARM side.

**Pull request merged**